pkg/cmd/tasks/deploy/predeploy.go

```diff
@@ -0,0 +1,79 @@
+package deploy
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/airplanedev/cli/pkg/api"
+	"github.com/airplanedev/cli/pkg/configs"
+	"github.com/airplanedev/cli/pkg/logger"
+	"github.com/airplanedev/cli/pkg/taskdir"
+	"github.com/airplanedev/cli/pkg/utils"
+	"github.com/pkg/errors"
+)
+
+// predeployHooks runs before a task is deployed and can e.g. prompt the user to specify or fix things
+func predeployHooks(ctx context.Context, client *api.Client, def taskdir.Definition) error {
+	// Check if configs exist
+	for k, v := range def.Env {
+		if v.Config != "" {
+			if err := ensureConfigExists(ctx, client, k, v.Config); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func ensureConfigExists(ctx context.Context, client *api.Client, envName, configName string) error {
+	cn, err := configs.ParseName(configName)
+	if err != nil {
+		return err
+	}
+	_, err = client.GetConfig(ctx, cn.Name, cn.Tag)
+	if err == nil {
+		return nil
+	}
+	switch err := errors.Cause(err).(type) {
+	case api.Error:
+		if err.Code != 404 {
+			return err
+		}
+		if !utils.CanPrompt() {
+			return errors.Errorf("config %s does not exist", configName)
+		}
+		logger.Log("Your task definition references config %s, which does not exist", logger.Bold(configName))
+		confirmed, errc := utils.Confirm("Create it now?")
```

Comment:
Any reason not to use `err` here?

```suggestion
		confirmed, err := utils.Confirm("Create it now?")
```

Reply:
Didn't want to shadow `err` above...

Reply:
Does this shadow err from the `return` below? Thought this would be inside the case statement's scope.

Reply:
`err` at this point is `api.Error` from the switch statement, so you get this error when comparing to `nil`:

![image](https://user-images.githubusercontent.com/64739/114819012-99ac4000-9d71-11eb-8911-ae5d15dd8ca4.png)